src/client/background.rs

```diff
@@ -0,0 +1,55 @@
+use futures::{Future, Poll};
+use hyper::body::Payload;
+use hyper::client::conn::Connection as HyperConnection;
+use log::error;
+use std::fmt::{self, Debug};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+/// Background task for a client connection.
+///
+/// This type is not used directly by a user of this library,
+/// but it can show up in trait bounds of generic types.
+pub struct Background<T, B>
+where
+    T: AsyncRead + AsyncWrite + Send + 'static,
+    B: Payload,
+{
+    connection: HyperConnection<T, B>,
```

Comment:
related to my other comment, we can put the LiftBody here in place of B which should be nice to hide the fact that we need to even lift the body.

Reply:
Done in 291d222.